cmd/influxd/run.go

```diff
@@ -567,17 +569,19 @@ func (cmd *RunCommand) openServer(joinURLs []url.URL) *influxdb.Server {
 	if err := s.Open(cmd.config.Data.Dir, c); err != nil {
 		log.Fatalf("failed to open data node: %v", err.Error())
 	}
-	log.Printf("data node opened at %s", cmd.config.Data.Dir)
+	log.Printf("data node(%d) opened at %s", s.ID(), cmd.config.Data.Dir)
 
-	dataNodeIndex := s.Index()
-	if dataNodeIndex == 0 {
+	// Give brokers time to elect a leader if entire cluster is being restarted.
+	time.Sleep(1 * time.Second)
```

Comment:
The data node currently requires the brokers to have a leader. On cluster restart, the first node comes up and tries to elect itself but can't because there's only one node. The second node comes up and sees that there's a broker so it tries to join on but gets a "not leader" message back and fails immediately. This means the second node's broker can't join in the consensus to elect a leader that the second node's data node needs.

By sleeping we give the broker time to reach consensus and elect a leader once the second node starts. This should be a retry but, again, this whole file needs rework. I'd rather not invest time fixing it right now when it's going to change soon.